### ISSUE TITLE
APL Code Question

### ISSUE THREAD

**Issue by sanavina** [simulationcraft/simc#4109]
Simulationcraft Version: 730-02

Can anyone explain what it means by "variable.s2mcheck" means in relation to this line of code:

surrender_to_madness,if=talent.surrender_to_madness.enabled&target.time_to_die<=variable.s2mcheck

From what I can tell, you use this ability if the target's time to die is less than or equal to ... 

I dont understand what you mean by the last part.

Thank you very much

**Comment by N1gh7h4wk** [comment]
"variable.s2mcheck" points to a variable that can be specified in the apl aswell, in this case the variable name is s2mcheck and its defined in the "check" action list of shadow priests
`actions.check+=/variable,op=set,name=s2mcheck,value=variable.s2msetup_time-(variable.actors_fight_time_mod*nonexecute_actors_pct)`

It basically calculates the time at which s2m should be triggered,
the variable.s2msetup_time is defined in the precombat action list as

`actions.precombat+=/variable,name=s2msetup_time,op=set,value=(0.8*(83+(20+20*talent.fortress_of_the_mind.enabled)*set_bonus.tier20_4pc-(5*talent.sanlayn.enabled)+((33-13*set_bonus.tier20_4pc)*talent.reaper_of_souls.enabled)+set_bonus.tier19_2pc*4+8*equipped.mangazas_madness+(raw_haste_pct*10*(1+0.7*set_bonus.tier20_4pc))*(2+(0.8*set_bonus.tier19_2pc)+(1*talent.reaper_of_souls.enabled)+(2*artifact.mass_hysteria.rank)-(1*talent.sanlayn.enabled)))),if=talent.surrender_to_madness.enabled`

tl;dr its a rough estimation of when s2m should be used

**Comment by lgkern** [comment]
The variable, as you deduced, is calculated at the start of the simulation to estimate when you should use Surrender to Madness. The reason for it to be that ugly expression is because Surrender to Madness has a lot of different factors that impact on its usages, ranging from talents to Tier bonuses. 
The expression tries to accomodate all those factors, instead of just having a flat value to when to use the talent.

**Comment by sanavina** [comment]
Thank you very much for the responses. That helps a lot!
